app/src/main/java/org/mozilla/rocket/shopping/search/ui/ShoppingSearchKeywordInputViewModel.kt

```diff
@@ -54,23 +55,22 @@ class ShoppingSearchKeywordInputViewModel(
         }
 
         fetchSuggestionsJob = viewModelScope.launch(Dispatchers.Default) {
-            if (TextUtils.isEmpty(keyword)) {
-                currentUiModel = currentUiModel.copy(keywordSuggestions = null, hideClear = true)
-            } else {
-                var styledSuggestions: List<CharSequence>? = null
-                val fetchKeywordSuggestionResult = fetchKeywordSuggestion(keyword)
-                if (fetchKeywordSuggestionResult is Result.Success) {
-                    styledSuggestions = applyStyle(keyword, fetchKeywordSuggestionResult.data)
-                }
-                currentUiModel = currentUiModel.copy(keywordSuggestions = styledSuggestions, hideClear = false)
+            var styledSuggestions: List<CharSequence>? = null
+            val fetchKeywordSuggestionResult = fetchKeywordSuggestion(keyword)
+            if (fetchKeywordSuggestionResult is Result.Success) {
+                styledSuggestions = applyStyle(keyword, fetchKeywordSuggestionResult.data)
             }
+            currentUiModel = currentUiModel.copy(
+                    keywordSuggestions = styledSuggestions,
+                    hideClear = TextUtils.isEmpty(keyword)
+            )
 
             withContext(Dispatchers.Main) {
```

Comment:
is this necessary?

Reply:
It's necessary since `currentUiModel` is assigned directly to `_uiModel.value`instead of using `postValue` in `emitUiModel()`